test/integration-tests/Uniswap.test.ts

```diff
@@ -1,235 +1,422 @@
+    })
+
+    it.skip('swap dai to weth', async () => {
+
+      const amountInDAI = expandTo18DecimalsBN(100);
+
+      let swapper = bob;
+      let swapperAddress = bob.address;
+
+      await daiContract.connect(swapper).approve(uniswapV2Router.address, MAX_UINT);
+      await wethContract.connect(swapper).approve(uniswapV2Router.address, MAX_UINT);
+
+      let amountIn = amountInDAI;
+      let amountOutMin = 0;
+      let path = [daiContract.address, wethContract.address];
+      let to = swapperAddress;
+      let deadlile = Math.round((new Date()).getTime() / 1000) + 86400;
+      await (await uniswapV2Router.connect(bob).swapExactTokensForTokens(
+        amountIn,
+        amountOutMin,
+        path,
+        to,
+        deadlile
+      )).wait();
+
     })
-    alice = await ethers.getSigner(ALICE_ADDRESS)
-    bob = (await ethers.getSigners())[1]
-    daiContract = new ethers.Contract(DAI.address, TOKEN_ABI, bob)
-    wethContract = new ethers.Contract(WETH.address, TOKEN_ABI, bob)
-    usdcContract = new ethers.Contract(USDC.address, TOKEN_ABI, bob)
-    permit2 = (await deployPermit2()).connect(bob) as Permit2
-    router = (await deployUniversalRouter(permit2)).connect(bob) as UniversalRouter
-    planner = new RoutePlanner()
-
-    // alice gives bob some tokens
-    await daiContract.connect(alice).transfer(bob.address, expandTo18DecimalsBN(100000))
-    await wethContract.connect(alice).transfer(bob.address, expandTo18DecimalsBN(100))
-
-    // Bob max-approves the permit2 contract to access his DAI and WETH
-    await daiContract.connect(bob).approve(permit2.address, MAX_UINT)
-    await wethContract.connect(bob).approve(permit2.address, MAX_UINT)
   })
 
   describe('Trade on Uniswap with Permit2, giving approval every time', () => {
     describe('ERC20 --> ERC20', () => {
-      let permit: PermitSingle
+      let permitSingle: PermitSingle;
+      beforeEach(async () => {
+        planner = new RoutePlanner();
+      })
 
       it('V2 exactIn, permiting the exact amount', async () => {
-        const amountInDAI = expandTo18DecimalsBN(100)
-        const minAmountOutWETH = expandTo18DecimalsBN(0.03)
+        const amountInDAI = expandTo18DecimalsBN(100);
+        const minAmountOutWETH = 0;
 
-        // second bob signs a permit to allow the router to access his DAI
-        permit = {
+        // second bob signs a permitSingle to allow the router to access his DAI
+        permitSingle = {
           details: {
-            token: DAI.address,
+            token: daiContract.address,
             amount: amountInDAI,
             expiration: 0, // expiration of 0 is block.timestamp
             nonce: 0, // this is his first trade
           },
           spender: router.address,
           sigDeadline: DEADLINE,
-        }
-        const sig = await getPermitSignature(permit, bob, permit2)
+        };
+
+        const signingKey: ethers.utils.SigningKey = new ethers.utils.SigningKey(BOB_PRIVATE_KEY);
```

Comment:
Overall, have you tried to use the Wallet and operate with Wallet instead of private keys?

Reply:
Can you try to do it, please? Together with simplified signature calculating, as I mentioned in another comment